### ISSUE TITLE
Build failure with gcc 9.2.0

### ISSUE THREAD

**Issue by ffontaine** [cameron314/concurrentqueue#218]
kismet embeds concurrentqueue since kismet-2020-09-R1 and https://github.com/kismetwireless/kismet/commit/397b590b7217378befc4ff8dc29035410fb44d7f. However, I got the following build failure is raised with gcc 9.2.0:

```
In file included from packetchain.h:45,
                 from packet.cc:36:
moodycamel/blockingconcurrentqueue.h: In constructor ‘moodycamel::BlockingConcurrentQueue<T, Traits>::BlockingConcurrentQueue(moodycamel::BlockingConcurrentQueue<T, Traits>::size_t, moodycamel::BlockingConcurrentQueue<T, Traits>::size_t, moodycamel::BlockingConcurrentQueue<T, Traits>::size_t)’:
moodycamel/blockingconcurrentqueue.h:68:133: error: no matching function for call to ‘moodycamel::BlockingConcurrentQueue<T, Traits>::create<moodycamel::BlockingConcurrentQueue<T, Traits>::LightweightSemaphore>(int, int)’
   68 |   : inner(minCapacity, maxExplicitProducers, maxImplicitProducers), sema(create<LightweightSemaphore>(0, (int)Traits::MAX_SEMA_SPINS), &BlockingConcurrentQueue::template destroy<LightweightSemaphore>)
      |                                                                                                                                     ^
moodycamel/blockingconcurrentqueue.h:555:19: note: candidate: ‘template<class T, class Traits> template<class U, class A1, class A2> static U* moodycamel::BlockingConcurrentQueue<T, Traits>::create(A1&&, A2&&)’
  555 |  static inline U* create(A1&& a1, A2&& a2)
      |                   ^~~~~~
moodycamel/blockingconcurrentqueue.h:555:19: note:   template argument deduction/substitution failed:
moodycamel/blockingconcurrentqueue.h:68:133: note:   mismatched types ‘A1’ and ‘int’
   68 |   : inner(minCapacity, maxExplicitProducers, maxImplicitProducers), sema(create<LightweightSemaphore>(0, (int)Traits::MAX_SEMA_SPINS), &BlockingConcurrentQueue::template destroy<LightweightSemaphore>)
      | 
```

This error can be fixed by reverting 4cc3bee07b9ef252cfdfd3fddbdb476dc750179f which casts `Traits::MAX_SEMA_SPINS` to `int` however it seems that this cast is needed with some other compilers. Do you have some insights and how this issue should be fixed?

**Comment by cameron314** [comment]
Looks like a bug in GCC: https://godbolt.org/z/r1ahE3
Seems to be fixed in 9.3.0.

I've committed a workaround. Let me know if it works.

**Comment by ffontaine** [comment]
Thanks being so quick in answering and committing a workaround.
